File: tinkoff/money.py
def getBanknotes(desiredSum, banknotes, sum=0, take=[]):
    if sum > desiredSum:
        return []
    if sum == desiredSum:
        return take
    for index in range(len(banknotes)):
        value = banknotes[index]
        sum1 = sum + value
        arr = take + [value]
        res = getBanknotes(desiredSum, banknotes[index + 1:], sum1, arr)
        if len(res) > 0:
            return res
    return []

desiredSum, banknoteCount = map(int, input().split())
banknotes = list(map(int, input().split()))
banknotes = banknotes + banknotes
banknotes.sort()
banknotes.reverse()

res = getBanknotes(desiredSum, banknotes)
if len(res):
  print(len(res))
  print(" ".join(map(str,res)))
else:
  print(-1)

File: tinkoff/money.ts
function getBanknotes(
  capacity: number,
  banknotes: number[],
  sum = 0,
  take: number[] = []
): number[] {
  if (sum > capacity) return [];
  if (sum == capacity) return take;
  for (let index in banknotes) {
    const value = banknotes[index];
    const sum1 = sum + value;
    const arr = [...take, value];
    const res = getBanknotes(capacity, banknotes.slice(+index + 1), sum1, arr);
    if (res.length) return res;
  }
  return [];
}

console.log(getBanknotes(5, [2,2,1,1]));
